rego/rego_test.go

```diff
@@ -2112,19 +2115,23 @@ p {
 // Catches issues around iteration with ND builtins.
 func TestNDBCacheWithRuleBodyAndIteration(t *testing.T) {
 	ctx := context.Background()
+	ts := httptest.NewServer(http.HandlerFunc(func(http.ResponseWriter, *http.Request) {
```

Comment:
Is there some unique identifier in each response, or how do we ensure that each of them will count in the set? I guess the date header would be one, but it's not too granular, is it?

Reply:
The sets are keyed with the arguments to the built-in. So we'll get three items because there are three different URLs used below.

Reply:
Hmm, is the `results` set not keyed by the responses?

Reply:
Yes it is. But the assertion checks some internals of NDBCache:
```
	// Ensure that the cache exists, and has exactly 3 entries.
	entries, ok := ndBC["http.send"]
	if !ok {
		t.Fatalf("expected http.send cache entry")
	}
	if entries.Len() != 3 {
		t.Fatalf("expected 3 http.send cache entries, received:\n%v", ndBC)
	}
```

Reply:
Ah, understood 👍 I imagined the tests were checking the actual set count in-policy. Should have read better :)